Portal/src/Datahub.Functions/TerraformOutputHandler.cs

```diff
@@ -391,6 +400,44 @@ private static string GetStatusMapping(string value)
                 $"Project resource not found for project acronym {projectAcronym.Value} and service type {terraformServiceType}");
         }
 
+
         return projectResource;
     }
+    private async Task<bool> DeleteProjectResource(IReadOnlyDictionary<string, TerraformOutputVariable> outputVariables, string terraformServiceType)
+    {
+        try {
+            
+            var projectAcronym = outputVariables[TerraformVariables.OutputProjectAcronym];
+        
+            var project = await _projectDbContext.Projects.Include(p => p.Resources)
+                                              .FirstOrDefaultAsync(p => p.Project_Acronym_CD == projectAcronym.Value);
+            if (project is null) {
+                _logger.LogError("Project not found for acronym {ProjectId}", projectAcronym.Value);
+                throw new Exception($"Project not found for acronym {projectAcronym.Value}");
+            }
+
+            var projectResource = project.Resources
+                .FirstOrDefault(x => x.ResourceType == terraformServiceType);
+
+            if (projectResource is null) {
+                _logger.LogError("Project resource not found for project acronym {ProjectAcronymValue} and service type {TerraformServiceType}",
+                                  projectAcronym.Value, terraformServiceType);
+                throw new Exception($"Project resource not found for project acronym {projectAcronym.Value} and service type {terraformServiceType}");
+            }
+
+            projectResource.Status = TerraformOutputStatus.Deleted;
+            _projectDbContext.Project_Resources2.Update(projectResource);
+
+            await _projectDbContext.SaveChangesAsync();
+
+            _logger.LogInformation("Project resource with service type {TerraformServiceType} deleted successfully for project acronym {ProjectAcronymValue}",
+                                    terraformServiceType, projectAcronym.Value);
+            return true;
+        }
+        catch (Exception e) {
+            _logger.LogError(e, "Error deleting project resource");
+        }
+
+        return true;
+    }
```

Comment:
In our previous call I suggested just putting L428 `projectResource.Status = storageBlobStatus` up on L277 instead of doing all this a second time inside a method, and fixing all the complex nesting.

Why was this not implemented?

Reply:
Apologies, for some reason, this particular comment was not visible in my review list. I've updated, will push in a munite

Reply:
That's not what I said, your update never sets the status unless it is deleted.

![image](https://github.com/ssc-sp/datahub-portal/assets/43834308/bbcc403a-94ce-42d1-8341-f8ea2c5ba319)

it should be something like

![image](https://github.com/ssc-sp/datahub-portal/assets/43834308/5ce12ab5-925c-4fec-8721-96035ee9db3a)

Reply:
No... why? Actually Status field would be used only in case of deleted... well mostly... unless you want to set the status field irrespective of deleted...

your code above is also correct - except that status field will always get set unless its either completed or deleted...
hmm... that amounts to the same thing too... well...

If we are going that route anyways, my thought is, probably, we can make it even more simple
i.e. always set the status field and let the output value from the terraform feed the status field

i.e. this would not be needed... coz it does not mean it has failed. It just means that the output status was set to some other value other than completed and deleted. So I believe we should not 'return' at this juncture, and let just set the status value be set. I think the value will be useful for later troubleshooting as well
![image](https://github.com/ssc-sp/datahub-portal/assets/19173124/4962ad32-d627-42e5-8b34-3f08a3a51af0)

What are your thoughts on that?

Reply:
like this 
![image](https://github.com/ssc-sp/datahub-portal/assets/19173124/29aa8f16-839d-4ff3-ad64-fbe5ba17dd34)

Reply:
In that case, it will set all values - including the deleted value